be/src/olap/rowset/segment_v2/ordinal_page_index.h

```diff
@@ -0,0 +1,131 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include "common/status.h"
+#include "olap/rowset/segment_v2/common.h"
+#include "olap/rowset/segment_v2/page_pointer.h"
+#include "util/coding.h"
+#include "util/slice.h"
+
+namespace doris {
+namespace segment_v2 {
+
+// this class encode ordinal page index
+// the binary format is like that
+// Header | Content
+// Header: 
+//      number of elements (4 Bytes)
+// Content:
+//      array of index_pair
+// index_pair:
+//      Ordinal (4 Bytes)
+//      PagePointer (8 Bytes)
+class OrdinalPageIndexBuilder {
+public:
+    OrdinalPageIndexBuilder() : _num_pages(0) {
+        _buffer.reserve(4 * 1024);
+        _buffer.resize(4);
```

Comment:
add a comment, eg: reserve the first 4 bytes for number of elements

Reply:
OK, I will fixed